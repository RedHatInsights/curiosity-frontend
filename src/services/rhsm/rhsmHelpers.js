import { RHSM_API_QUERY_SET_TYPES } from './rhsmConstants';

/**
 * ToDo: remove filterArchitectureVariant when the API supports architecture, variant params
 */
/**
 * Patch for returning a made up API architecture, variant param as a product ID
 *
 * @param {string} id
 * @param {object} params
 * @returns {string}
 */
const filterArchitectureVariant = (id, params = {}) => {
  const updatedId = id;

  if (params?.[RHSM_API_QUERY_SET_TYPES.ARCHITECTURE]?.length) {
    return params?.[RHSM_API_QUERY_SET_TYPES.ARCHITECTURE];
  }

  if (params?.[RHSM_API_QUERY_SET_TYPES.VARIANT]?.length) {
    return params?.[RHSM_API_QUERY_SET_TYPES.VARIANT];
  }

  return updatedId;
};

const rhsmHelpers = {
  filterArchitectureVariant
};

export { rhsmHelpers as default, rhsmHelpers, filterArchitectureVariant };
