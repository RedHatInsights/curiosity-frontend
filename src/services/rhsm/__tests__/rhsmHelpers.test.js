import { rhsmHelpers, billingMetrics } from '../rhsmHelpers';

describe('RhsmHelpers', () => {
  it('should have specific functions', () => {
    expect(rhsmHelpers).toMatchSnapshot('rhsmHelpers');
  });

  it.each([
    {
      description: 'basic',
      data: undefined
    },
    {
      description: 'multiple unique providers, multiple accounts',
      data: {
        loremApi: [
          { id: '123', provider: 'ipsum', type: 'loremApi' },
          { id: '456', provider: 'lorem', type: 'loremApi' },
          { id: '789', provider: 'lorem', type: 'loremApi' },
          { id: '012', provider: 'dolor', type: 'loremApi' }
        ],
        dolorApi: [{ id: '789', provider: 'lorem', type: 'dolorApi' }]
      }
    },
    {
      description: 'single provider, multiple accounts',
      data: {
        loremApi: [
          { id: '123', provider: 'lorem', type: 'loremApi' },
          { id: '456', provider: 'lorem', type: 'loremApi' },
          { id: '789', provider: 'lorem', type: 'loremApi' },
          { id: '012', provider: 'lorem', type: 'loremApi' }
        ],
        dolorApi: [{ id: '789', provider: 'lorem', type: 'dolorApi' }]
      }
    },
    {
      description: 'single unique provider, multiple accounts',
      data: {
        loremApi: [
          { id: '123', provider: 'lorem', type: 'loremApi' },
          { id: '456', provider: 'lorem', type: 'loremApi' },
          { id: '789', provider: 'dolor', type: 'loremApi' },
          { id: '012', provider: 'lorem', type: 'loremApi' }
        ],
        dolorApi: [{ id: '789', provider: 'dolor', type: 'dolorApi' }]
      }
    },
    {
      description: 'single provider, single account',
      data: {
        loremApi: [
          { id: '123', provider: 'lorem', type: 'loremApi' },
          { id: '789', provider: 'lorem', type: 'loremApi' }
        ],
        dolorApi: [{ id: '789', provider: 'lorem', type: 'dolorApi' }]
      }
    },
    {
      description: 'single unique provider, single account',
      data: {
        loremApi: [
          { id: '123', provider: 'lorem', type: 'loremApi' },
          { id: '789', provider: 'dolor', type: 'loremApi' }
        ],
        dolorApi: [{ id: '789', provider: 'dolor', type: 'dolorApi' }]
      }
    }
  ])('billingMetrics should return parsed API responses in a consumable metric format, $description', ({ data }) => {
    expect(billingMetrics(data)).toMatchSnapshot();
  });
});
