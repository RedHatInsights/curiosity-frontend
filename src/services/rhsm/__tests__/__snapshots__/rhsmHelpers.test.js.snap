// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RHSM Helpers filterArchitectureVariant, should filter architectures, variants from an object: passed id 1`] = `"passThroughId"`;

exports[`RHSM Helpers filterArchitectureVariant, should filter architectures, variants from an object: updated id, architecture 1`] = `"lorem"`;

exports[`RHSM Helpers filterArchitectureVariant, should filter architectures, variants from an object: updated id, variant 1`] = `"ipsum"`;

exports[`RHSM Helpers should have specific functions: helpers 1`] = `
{
  "filterArchitectureVariant": [Function],
}
`;
